src/PFRICH_geo_v1.cpp

```diff
@@ -0,0 +1,1407 @@
+   float clearance = 0.5;                   // in cm
+   float length = 45;                       // in cm
+
+
+/////*--------------------------------------------------*/
+//  /// Inner mirror cone
+//  //
+//  // FIXME: do I really care about re-using the same names for these shapes?;
+//  // A wedge bridging two cylinders;
+
+  Tube   eflange(0.0, _FLANGE_EPIPE_DIAMETER_/2 + clearance, length/2);
+  Tube   hflange(0.0, _FLANGE_HPIPE_DIAMETER_/2 + clearance, length/2);
+
+  double r0 = _FLANGE_EPIPE_DIAMETER_/2 + clearance;
+  double r1 = _FLANGE_HPIPE_DIAMETER_/2 + clearance;
+  double L = _FLANGE_HPIPE_OFFSET_;
+  double a = r0*L/(r0-r1);
+  double b = r0*r0/a;
+  double c = r1*(a-b)/r0;
+
+  // GEANT variables to define G4Trap;
+  double pDz = length/2, pTheta = 0.0, pPhi = 0.0, pDy1 = (a - b - c)/2, pDy2 = pDy1;
+  double pDx1 = sqrt(r0*r0 - b*b), pDx2 = pDx1*r1/r0, pDx3 = pDx1, pDx4 = pDx2, pAlp1 = 0.0, pAlp2 = 0.0;
+
+  Trap wedge(pDz, pTheta, pPhi, pDy1, pDx1, pDx2, pAlp1, pDy2, pDx3, pDx4, pAlp2);
+
+  // flange_shape = new G4UnionSolid("Tmp", eflange, hflange, 0, G4ThreeVector(-_FLANGE_HPIPE_OFFSET_, 0.0, 0.0));
+//  Rotation3D  rot(RotationZYX(0, M_PI, 0));
+//  Transform3D transform(rot, Position(0, 0, -149));
+
+  UnionSolid flange_shape(eflange, hflange, Position(-_FLANGE_HPIPE_OFFSET_, 0.0, 0.0));
+
+  Rotation3D rZ(RotationZYX(M_PI/2.0, 0.0, 0.0));
+
+  Transform3D transform_flange(rZ, Position(-b - pDy1, 0.0, 0.0));
+
+  UnionSolid flange_final_shape(flange_shape, wedge, transform_flange);
+
+//  Volume flangeVol(detName + "_flange", flange_shape, mirrorMat);
+  Volume flangeVol(detName + "_flange", flange_final_shape, mirrorMat);
+  flangeVol.setVisAttributes(mirrorVis);
+
+   PlacedVolume flangePV = volume.placeVolume(flangeVol, Position(0, 0, 0));
+   DetElement   flangeDE(sdet, "flange_de", 0);
+   flangeDE.setPlacement(flangePV);
+
+  ///*--------------------------------------------------*/
+  /// SENSOR MODULE LOOP ------------------------
+  /* cartesian tiling loop
+   * - start at (x=0,y=0), to center the grid
+   * - loop over positive-x positions; for each, place the corresponding negative-x sensor too
+   * - nested similar loop over y positions
+   */
+
+   Box    sensorSolid(sensorSide / 2., sensorSide / 2., sensorThickness / 2.);
+   Volume sensorVol(detName + "_sensor", sensorSolid, sensorMat);
+   sensorVol.setVisAttributes(sensorVis);
+
+   // -- Mirrors ---------------------------------------------------------------------------------
+   //
+   // Some "standard" value applied to all mirrors;
+   double _MIRROR_REFLECTIVITY_ = 0.90;
+
+   // At the downstream (sensor plane) location; upstream radii are calculated automatically;
+   double _CONICAL_MIRROR_INNER_RADIUS_ = 12.0;
+   double _CONICAL_MIRROR_OUTER_RADIUS_ = 57.0;
+
+   double _INNER_MIRROR_THICKNESS_ = 0.1;  //0.29*_INCH
+   double _OUTER_MIRROR_THICKNESS_ = 0.2;  //0.54*_INCH
+
+//   #ifdef _PLANACON_GEOMETRY_
```

Comment:
Configuration with `ifdef` should be avoided. We distribute compiled code.

Reply:
Fixed